docs/sql-data-sources-load-save-functions.md

```diff
@@ -41,6 +41,11 @@ name (i.e., `org.apache.spark.sql.parquet`), but for built-in sources you can al
 names (`json`, `parquet`, `jdbc`, `orc`, `libsvm`, `csv`, `text`). DataFrames loaded from any data
 source type can be converted into other types using this syntax.
 
+For built-in sources, the available extra options are documented in the API documentation,
```

Comment:
BTW, I asked https://github.com/apache/spark/pull/23742#discussion_r254673374 for the reasons below:

- documented, documentation sounds duplicated
- `the method corresponding to the format` some formats don't have the methods. Like avro, kafka, etc.
- There are more places like `org.apache.spark.sql.DataStreamReader`.

So it better makes it simpler and generalized.

Reply:
@HyukjinKwon can you please summarise what does "_There are more places like `org.apache.spark.sql.DataStreamReader`" mean: where are those properties documented?

I think it would be important to capture at least the places one should be looking at, not just a vague reference that "please see the API docs for the details"

If you put yourself into the shoes of a new Spark developer, who has no understanding of the internal classes, configuring data loading/saving/ properly becomes a hard task.

For the `the method corresponding to the format`: I believe avro and kafka are not built-in by default are they? So that statement is correct isn't it?

Reply:
vague but we shouldn't whitelist here for now. Otherwise, we should fix the documentation every time we add an API. Simply pointing out an API should be good enough for now.

To permanently resolve the issue, we should find a way to deduplicate the doc and avoid maintenance overhead. If you're going to deal with this, please go ahead.

For instance, libsvm is a builtin source but does not have `libsvm` method. Shall we avoid to whitelist here to avoid maintenance overhead?

Reply:
OK, that makes sense. I've updated the paragraph with that. I do think it makes sense to point people to actual class names at least for the basic formats. 

The content of the last commit should explain the situation well enough.